web/src/controllers/places.rs

```diff
@@ -126,6 +141,14 @@ pub async fn search(
     Json(place_req): Json<OsdmPlaceRequest>,
 ) -> PlacesResponse {
     let maybe_place_input = place_req.place_input;
+    let maybe_restrictions = place_req.restrictions;
+
+    let limit = match maybe_restrictions {
+        Some(restrictions) => restrictions
+            .number_of_results
+            .unwrap_or(DEFAULT_NUMBER_OF_RESULTS),
+        None => DEFAULT_NUMBER_OF_RESULTS,
+    };
```

Comment:
Alternatively:

```suggestion
    let limit = maybe_restrictions
        .map(|restrictions| restrictions.number_of_results)
        .flatten()
        .unwrap_or(DEFAULT_NUMBER_OF_RESULTS);
```

Perhaps too clever, but also may depend on your IDE. My current set up tells me what the type is at the end of each line, making this style very clear (to me).

Reply:
It's an improvement, we only use the constant once 👍

Reply:
Oh, that's indeed better :+1: